src/backend/backend.ts

```diff
@@ -229,26 +217,34 @@ const resubscribe = () => {
     parsedModulesData.tokenIdMap = {};
 
     setTimeout(() => {
-      Promise.resolve().then(() => {
-        runAndHandleUncaughtExceptions(() => {
-          const roots = collectRoots();
-          if (roots.length) {
-            const appRef: ApplicationRef = parseModulesFromRootElement(roots[0], parsedModulesData);
-            if (isStableSubscription) { isStableSubscription.unsubscribe(); }
-            isStableSubscription = appRef.isStable.subscribe((e) => {
-              updateComponentTree(collectRoots());
-              updateRouterTree();
-              send(MessageFactory.ping());
-            });
-            ngModuleRef = (appRef as any)._injector;
-            ngModuleRef.onDestroy(() => {
-              ngModuleRef = undefined;
-              listenForSomeTimeAndMaybeResubscribe(1000);
-            });
-            sendNgModulesMessage();
-          }
-        });
-      })
+      Promise.resolve()
+        .then(() => {
+          runAndHandleUncaughtExceptions(() => {
+            const roots = collectRoots();
+            if (roots.length) {
+              let sanity;
+              const sanityThreshold = 0.5 * 1000; // 0.5 seconds
```

Comment:
@feeloor Please add a comment explaining what is going on. Rest looks good to me.

Reply:
@sumitarora I've now added a few comments. 👍